sdk/formrecognizer/azure-ai-formrecognizer/samples/async_samples/sample_differentiate_custom_forms_with_labeled_and_unlabeled_models_async.py

```diff
@@ -0,0 +1,97 @@
+# coding: utf-8
+
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+"""
+FILE: sample_differentiate_custom_forms_with_labeled_and_unlabeled_models_async.py
+
+DESCRIPTION:
+    This sample demonstrates the differences in output that arise when recognize_custom_forms
+    is called with custom models trained with labeled and unlabeled data.
+USAGE:
+    python sample_differentiate_custom_forms_with_labeled_and_unlabeled_models_async.py
+
+    Set the environment variables with your own values before running the sample:
+    1) AZURE_FORM_RECOGNIZER_ENDPOINT - the endpoint to your Cognitive Services resource.
+    2) AZURE_FORM_RECOGNIZER_KEY - your Form Recognizer subscription key
+    3) CUSTOM_TRAINED_LABELED_MODEL_ID - the ID of your custom model trained with labeled data
+    4) CUSTOM_TRAINED_UNLABELED_MODEL_ID - the ID of your custom model trained with unlabeled data
+"""
+
+import os
+import asyncio
+
+
+class DifferentiateCustomFormsWithLabeledAndUnlabeledModelsAsync(object):
+
+    endpoint = os.environ["AZURE_FORM_RECOGNIZER_ENDPOINT"]
+    key = os.environ["AZURE_FORM_RECOGNIZER_KEY"]
+    labeled_model_id = os.environ["CUSTOM_TRAINED_LABELED_MODEL_ID"]
+    unlabeled_model_id = os.environ["CUSTOM_TRAINED_UNLABELED_MODEL_ID"]
+
+    async def recognize_custom_forms(self):
+        from azure.core.credentials import AzureKeyCredential
+        from azure.ai.formrecognizer.aio import FormRecognizerClient
+        async with FormRecognizerClient(
+            endpoint=self.endpoint, credential=AzureKeyCredential(self.key)
+        ) as form_recognizer_client:
+
+            # The form you are recognizing must be of the same type as the forms the custom model was trained on
+            with open("sample_forms/forms/Form_1.jpg", "rb") as f:
+                stream = f.read()
+            forms_with_labeled_model = await form_recognizer_client.recognize_custom_forms(
+                model_id=self.labeled_model_id, stream=stream
+            )
+            forms_with_unlabeled_model = await form_recognizer_client.recognize_custom_forms(
+                model_id=self.unlabeled_model_id, stream=stream
```

Comment:
interesting: you can re-use the stream?

Reply:
Yeah I guess so, I didn't think about that being weird, but it works in my sample